libyul/backends/wasm/EVMToEWasmTranslator.cpp

```diff
@@ -0,0 +1,82 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/**
+ * Translates Yul code from EVM dialect to eWasm dialect.
+ */
+
+#include <libyul/backends/wasm/EVMToEWasmTranslator.h>
+
+#include <libyul/backends/wasm/WordSizeTransform.h>
+#include <libyul/backends/wasm/WasmDialect.h>
+#include <libyul/optimiser/ExpressionSplitter.h>
+
+#include <libyul/AsmParser.h>
+
+#include <liblangutil/ErrorReporter.h>
+#include <liblangutil/Scanner.h>
+
+using namespace std;
+using namespace dev;
+using namespace yul;
+using namespace langutil;
+
+namespace
+{
+static string const polyfill{R"({
```

Comment:
We finally made it through the rabbit hole and out the other side!
@wanderer - some months (years?) ago, I think you already wrote those translation functions (emulate EVM instructions in WebAssembly) in Yul (then still called julia I think) do you still have that code by any chance? :)

Reply:
They are here, but they are not useful: https://github.com/axic/julia-wasm-arithmetics

The better implementation is in https://github.com/ewasm/evm2wasm/tree/master/wasm

But you'd be better off writing it from scratch.